libraries/tock-cells/src/map_cell.rs

```diff
@@ -86,6 +86,23 @@ impl<T> MapCell<T> {
         }
     }
 
+    /// Sets the `MapCell` value.
+    ///
+    /// If the passed value is a `Some(T)`, the MapCell will be set to
+    /// be occupied with this value. If the passed value is a `None`,
+    /// the `MapCell` will not be occupied.
+    ///
+    /// The previous value will be returned.
+    pub fn set(&self, val: Option<T>) -> Option<T> {
```

Comment:
I guess that would work, but it's not going to be a single-method replacement of `OptionalCell::insert` as previously used, but rather requires the expression `if let Some(cb) = callback { self.callback.replace(cb) } else { self.callback.take() }`. This is really only required for the remaining Tock 1.x capsules, as in 2.0 those `OptionalCell`s would be replaced by `RefCell`s accordingly. Would you be fine integrating this temporarily, potentially under another name?

Reply:
I am fine with it temporarily under a different name -- maybe `replace_opt()`?

Reply:
I have removed the offending commit from the patchset. Since there were not too many capsules still using `LegacyDriver`, migrating these to the `if let` expression wasn't too bad.